File: app/src/main/java/com/github/mrvilkaman/presentationlayer/fragments/testfrags/Frag1ScreenFragment.java
package com.github.mrvilkaman.presentationlayer.fragments.testfrags;

import com.github.mrvilkaman.di.ActivityCoreComponent;
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment;

public class Frag1ScreenFragment extends FragBaseScreenFragment {


	public static Frag1ScreenFragment open() {
		return new Frag1ScreenFragment();
	}

	@Override
	protected int getNumber() {
		return 1;
	}

	@Override
	protected BaseFragment nextFragment() {
		return Frag2ScreenFragment.open();
	}

	@Override
	public void daggerInject() {
		ActivityCoreComponent component = getComponent(ActivityCoreComponent.class);
		DaggerFragScreenComponent.builder().activityCoreComponent(component).build().inject(this);
	}
}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/Frag1ScreenFragment.kt
package com.github.mrvilkaman.ui.screens.testfrags

import com.github.mrvilkaman.ui.screens.ScreenKey

class Frag1ScreenFragment : FragBaseScreenFragment() {

    override val number: Int
        get() = 1

    override fun nextScreenKey(): String? = ScreenKey.FRAG2


    companion object {
        fun open(): Frag1ScreenFragment {
            return Frag1ScreenFragment()
        }
    }
}
